### PR TITLE
feat(confirm): promote "open gmail" to a feature

### LINKED ISSUE #3368: Enable "Open Gmail" for Desktop @gmail users

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #3368

### CODE REVIEW

**Comment by vladikoff** [comment]
Fixes #3368

**Comment by rfk** [comment]
Look at all those beautiful red code-deletion lines!  I don't feel qualified to detail-review this, but :+1:

File: app/scripts/views/confirm.js [hunk 1]
```diff
@@ -53,7 +49,9 @@ define(function (require, exports, module) {
     },
 
     _isOpenGmailButtonVisible: function () {
-      return this.isInExperimentGroup('openGmail', 'treatment');
+      // The "Open Gmail" is only visible in certain contexts
+      // we do not show it in mobile contexts because it performs worse on mobile
```
Comment (shane-tomlinson) [inline app/scripts/views/confirm.js]: great explanation.

File: app/tests/spec/views/confirm.js [hunk 2]
```diff
@@ -363,7 +352,39 @@ define(function (require, exports, module) {
...
+      });
+    });
+
+    describe('openGmail feature', function () {
+      it('it is not visible in basic contexts', function () {
+        assert.notOk($('#open-gmail').length);
+      });
+
+      it('it is visible in desktop contexts', function () {
+        broker = new FxDesktopV3AuthenticationBroker({
```
Comment (shane-tomlinson) [inline app/tests/spec/views/confirm.js]: You don't actually need the FxDesktopV3AuthenticationBroker here, and it's actually a little brittle to do so. You could use a BaseBroker instance, add the capability, e.g., `broker.setCapability('openGmailButtonVisible', true);` then run the rest of the test.

I'd also rename the test to something like "is visible if the broker supports the `openGmailButtonVisible` capability"  or similar which is what we really want.

File: app/tests/spec/views/confirm.js [hunk 3]
```diff
@@ -363,7 +352,39 @@ define(function (require, exports, module) {
...
+          relier: relier,
+          user: user,
+          viewName: 'confirm',
+          window: windowMock
+        });
+
+        return view.render()
+          .then(function () {
+            $('#container').html(view.el);
+            assert.ok($('#open-gmail').length);
```
Comment (shane-tomlinson) [inline app/tests/spec/views/confirm.js]: You can do an `assert.lengthOf(view.$('#open-gmail'), 1)` - I just learned about Chai's [assert.lengthOf](http://chaijs.com/api/assert/#lengthOf) test recently.

File: tests/functional/confirm.js [hunk 4]
```diff
@@ -61,6 +67,48 @@ define([
...
         .then(FunctionalHelpers.testSuccessWasShown(this));
+    },
+
+    'sign up with a gmail address, get the open gmail button': function () {
+      var self = this;
+      var email = 'signin' + Math.random() + '@gmail.com';
+      var PASSWORD = '12345678';
+      var SIGNUP_URL = intern.config.fxaContentRoot + 'signup?context=fx_desktop_v2&service=sync';
+
+      return this.remote
```
Comment (shane-tomlinson) [inline tests/functional/confirm.js]: Can you use `FunctionalHelpers.openPage(SIGNUP_URL, '#fxa-signup-header')` to open the page? It'll reduce the boilerplate, plus allow us to do other customizations like hook up the `listenForFxaCommands` helper universally.

**Comment by shane-tomlinson** [comment]
I consistently see this functional test failure:

>× firefox on any platform - sign_up - sign up with a verified account forces the user to sign in (1.621s)
AssertionError: expected '' to equal 'signin0.8223928271327168@restmail.net'
  at Function.assert.equal  <node_modules/intern/node_modules/chai/chai.js:2028:10>
  at Command.<anonymous>  <tests/functional/sign_up.js:317:24>
  at runCallback  <node_modules/leadfoot/Command.js:526:31>
  at Command.<anonymous>  <node_modules/leadfoot/Command.js:543:11>
  at <node_modules/dojo/Promise.ts:393:15>
  at run  <node_modules/dojo/Promise.ts:237:7>
  at <node_modules/dojo/nextTick.ts:44:3>
  at doNTCallback0  <node.js:419:9>
  at process._tickCallback  <node.js:348:13>
  at Command.then  <node_modules/leadfoot/Command.js:542:10>
  at <tests/functional/sign_up.js:315:16>
  at onSettled  <app/bower_components/fxa-js-client/fxa-client.js:1312:9>
  at Immediate.flush [as _onImmediate]  <app/bower_components/fxa-js-client/fxa-client.js:1019:6>
  at processImmediate [as _immediateCallback]  <timers.js:383:17>

I only see this when running all of the sign_up tests, not when run in isolation.

**Comment by shane-tomlinson** [comment]
> I only see this when running all of the sign_up tests, not when run in isolation.

I take this back, I only see this when running all the tests. :/

**Comment by shane-tomlinson** [comment]
> I take this back, I only see this when running all the tests. :/

The test failures are caused by a timing issue with Selenium Standalone Server 2.50.1, see #3461.

**Comment by shane-tomlinson** [comment]
@vladikoff - aside from my nits, this PR looks great! With Selenium 2.47.1, all the tests pass. Merge when nits addressed.

File: app/tests/spec/views/confirm.js (hunk 2 above)
Comment (vladikoff) [inline app/tests/spec/views/confirm.js]: ah yes, thanks

File: app/tests/spec/views/confirm.js (hunk 3 above)
Comment (vladikoff) [inline app/tests/spec/views/confirm.js]: :+1:

**Pull request merged**